OpenRA.Mods.Common/Widgets/RadarWidget.cs

```diff
@@ -176,24 +176,21 @@ void MapBoundsChanged()
 
 			// The minimap is drawn in cell space, so we need to
 			// unproject the bounds to find the extent of the map.
+			// TODO: This attempt to find the map bounds accounting for projected cell heights is bogus.
+			// When a map with height is involved, the bounds may not be optimal, this needs fixing.
 			var projectedLeft = map.Bounds.Left;
 			var projectedRight = map.Bounds.Right;
 			var projectedTop = map.Bounds.Top;
-			var projectedBottom = map.Bounds.Bottom;
 			var top = int.MaxValue;
-			var bottom = int.MinValue;
-			var left = map.Bounds.Left * cellWidth;
-			var right = map.Bounds.Right * cellWidth;
+			var bottom = map.Bounds.Bottom;
+			var left = projectedLeft * cellWidth;
+			var right = projectedRight * cellWidth;
 
 			for (var x = projectedLeft; x < projectedRight; x++)
 			{
 				var allTop = map.Unproject(new PPos(x, projectedTop));
-				var allBottom = map.Unproject(new PPos(x, projectedBottom));
 				if (allTop.Count > 0)
 					top = Math.Min(top, allTop.MinBy(uv => uv.V).V);
-
-				if (allBottom.Count > 0)
-					bottom = Math.Max(bottom, allBottom.MaxBy(uv => uv.V).V);
```

Comment:
we could keep the code, just add an else statement to 
```
if (allBottom.Count > 0)
	bottom = Math.Max(bottom, allBottom.MaxBy(uv => uv.V).V);
else
	bottom = map.Bounds.Bottom;

Reply:
Updated.

Reply:
Perhaps add a comment to `allBottom`
```
// If every tile along the bottom edge has a height > 0 `Map.Unproject` returns an empty list
``` 
so it's clearer in the future what needs fixing. Also couldn't this bug also happen to `allTop`?

Reply:
Added a longer explanatory comment above it all.

Getting top to fail would be harder, but it's possible to imagine a narrow map set to max height would have the entire projected space be offset enough that none of it overlaps with the map space. So added a guard there too.

This is of course all still bogus garbage in the first place so keeping the TODO comment, but at least it's garbage that doesn't crash now.